frame.go

```diff
@@ -952,6 +955,10 @@ func (f *framer) parsePreparedMetadata() preparedMetadata {
 	// TODO: deduplicate this from parseMetadata
 	meta := preparedMetadata{}
 
+	if f.proto > protoVersion4 {
```

Comment:
I'm sorry for the misunderstanding here. The `id` field in `preparedMetadata` doesn't represent `<id>` in the specification, but `result_metadata_id`. 
Also, the naming is terrible here, which leads to a big misunderstanding.

I moved parsing of the `result_metadata_id` from this method to the `parseResultPrepared` method, because the `result_metadata_id` field is a part of the **PREPARED** message, not its metadata.